hack/ci/cleanup_machines.sh

```diff
@@ -17,25 +17,23 @@
 set -euo pipefail
 set -x
 
-cd $(dirname $0)
+# This ensures that we are only performing a delete if the cluster was exposed. If it wasn't exposed
+# then we never created any machines. So, cleanup is not required.
+if [ -f ~/.kube/config-external ] && [ ! -z "${PROW_JOB_ID:-}" ]; then
```

Comment:
`! -z` == `-n`

I am a bit unsure why this stopgap mechanism is needed. It says "if this file exists and we're in a Prowjob, then exit.", which doesn't make sense to me. 🤔

Reply:
Updated the check, it should have been `If the file doesn't exist and we're running this in a Prowjob, then we don't need any cleanup`. TBH just the first part, checking for the existence of the external Kubeconfig should be sufficient.